packages/@uppy/thumbnail-generator/src/index.ts

```diff
@@ -223,40 +307,45 @@ export default class ThumbnailGenerator extends UIPlugin {
     let sH = targetHeight * 2 ** (steps - 1)
     const x = 2
 
+    let canvas: HTMLCanvasElement
+
     while (steps--) {
-      const canvas = document.createElement('canvas')
+      canvas = document.createElement('canvas')
       canvas.width = sW
       canvas.height = sH
-      canvas.getContext('2d').drawImage(img, 0, 0, sW, sH)
-      img = canvas
+      canvas.getContext('2d')!.drawImage(img, 0, 0, sW, sH)
```

Comment:
This doesn't look like a safe change, can you revert it?

Reply:
We were reusing a variable too much with different types. No harm in cleaning up while we're there.

Reply:
The code is not bad though – or at least, it was doing what we expect when your proposal does not